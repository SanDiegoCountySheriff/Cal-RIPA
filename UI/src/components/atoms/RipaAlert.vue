<template>
  <v-alert :outlined="alertOutlined" :type="alertType" dense>
    <slot></slot>
  </v-alert>
</template>

<script>
export default {
  name: 'ripa-alert',

  props: {
    alertType: {
      type: String,
      default: 'info',
    },
    alertOutlined: {
      type: Boolean,
      default: false,
    },
  },
}
</script>
