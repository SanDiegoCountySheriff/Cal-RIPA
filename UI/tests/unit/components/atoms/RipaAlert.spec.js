import { mount } from '@vue/test-utils'
import RipaAlert from '@/components/atoms/RipaAlert.vue'
import Vuetify from 'vuetify'

describe('Ripa Alert', () => {
  let vuetify
  let wrapper = null

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = (propsData, slotsData = '') => {
    return mount(RipaAlert, {
      vuetify,
      propsData: {
        ...propsData,
      },
      slots: {
        default: slotsData,
      },
    })
  }

  it('should display the default alert type and outline', () => {
    wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display warning and default outline', () => {
    wrapper = factory({ alertType: 'warning' })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display outline', () => {
    wrapper = factory({ alertOutlined: true })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should display slot data', () => {
    wrapper = factory({}, 'Test Slot Data')
    expect(wrapper.html()).toContain('Test Slot Data')
  })
})
