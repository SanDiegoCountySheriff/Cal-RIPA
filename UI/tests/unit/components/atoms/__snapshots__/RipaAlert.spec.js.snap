// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Alert should display outline 1`] = `<v-alert-stub outlined="true" tag="div" value="true" closelabel="$vuetify.close" dense="true" closeicon="$cancel" icon="" type="info"></v-alert-stub>`;

exports[`Ripa Alert should display the default alert type and outline 1`] = `<v-alert-stub tag="div" value="true" closelabel="$vuetify.close" dense="true" closeicon="$cancel" icon="" type="info"></v-alert-stub>`;

exports[`Ripa Alert should display warning and default outline 1`] = `<v-alert-stub tag="div" value="true" closelabel="$vuetify.close" dense="true" closeicon="$cancel" icon="" type="warning"></v-alert-stub>`;
